database/clickhouse/clickhouse.go

```diff
@@ -216,14 +225,19 @@ func (ch *ClickHouse) ensureVersionTable() (err error) {
 	} else {
 		return nil
 	}
+
 	// if not, create the empty migration table
-	query = `
-		CREATE TABLE ` + ch.config.MigrationsTable + ` (
-			version    Int64, 
+	query = fmt.Sprintf(`
+		CREATE TABLE %s (
+			version    Int64,
 			dirty      UInt8,
 			sequence   UInt64
-		) Engine=TinyLog
-	`
+		) Engine=%s`, ch.config.MigrationsTable, ch.config.TableEngine)
+
+	if strings.HasSuffix(ch.config.TableEngine, "Tree") {
+		query = fmt.Sprintf(`%s ORDER BY sequence`, query)
```

Comment:
Why sort by `sequence` instead of `version` or `(version, sequence)`?

Reply:
No particular strong reason, but the sequence is the natural progression order of the migration versions and the corresponding state at the time. For this particular table size, access frequency and pattern it doesn't make much difference.

We can change it, but it seems counter intuitive for me go for version and then sequence.

Let me know if we need to change it.

Reply:
Ahh, `sequence` makes sense since it'll be the primary key. I forgot that the clickhouse db driver tracks all applied migrations and only looks at the latest one for the current state

Reply:
Yes :)